src/viam/sdk/rpc/dial.hpp

```diff
@@ -117,12 +120,32 @@ class Options {
         : refresh_interval_(std::move(refresh_interval)), dial_options_(std::move(dial_options)) {}
 
     unsigned int refresh_interval() const;
+    unsigned int check_every_interval() const;
+    unsigned int reconnect_every_interval() const;
+
+    /// @brief Sets the frequency (in seconds) to verify connectivity
+    Options& set_check_every_interval(unsigned int interval);
+
+    /// @brief Sets the frequency (in seconds) to attempt to reconnect when connectivity is lost
+    Options& set_reconnect_every_interval(unsigned int interval);
     const boost::optional<DialOptions>& dial_options() const;
 
    private:
     /// @brief How often to refresh the status/parts of the robot, in seconds. If set to 0, the
     /// robot will not automatically refresh.
     unsigned int refresh_interval_;
+
+    /// @brief How often to verify connectivity to the robot, in seconds. If set to 0, will not
+    /// check, will default to the `reconnect_every_interval_` value. Defaults to 0.
+    /// @note Setting to a non-zero value is useful in modules but may result in delays shutting
+    /// down client code
+    unsigned int check_every_interval_ = 0;
+
+    /// @brief How often to attempt to reconnect to the robot when disconnected. If set to 0,
+    /// will not attempt to reconnect. Defaults to 0.
+    /// @note Setting to a non-zero value is useful in modules but may result in delays shutting
+    /// down client code
+    unsigned int reconnect_every_interval_ = 0;
```

Comment:
I opted to set these both to `0` by default. It leads to no meaningful behavioral change for existing clients, and sidesteps the issue of having to wait for the `check_connection` loop to finish before shutting down client code. Also if a client is having issues connecting, they probably can figure out what they want to do and stop the process as necessary by interacting directly, without us making any assumptions.

this is overridden for modules in `modules/service.cpp`, to ensure that modules don't get orphaned.